File: 2024/python/01_first_gift_repeated.py
"""01. First Gift Repeated!"""


def prepare_gifts(gifts: list) -> list:
    """
    Prepare gifts.

    Returns a new list without duplicates and sorted in ascending order.

    Args:
        gifts (list): The list of gifts to prepare

    Returns:
        list: A list with the prepared gifts
    """
    return sorted(list(set(gifts)))


gifts1 = [3, 1, 2, 3, 4, 2, 5]
prepared_gifts_1 = prepare_gifts(gifts1)
print(prepared_gifts_1)  # [1, 2, 3, 4, 5]

gifts2 = [6, 5, 5, 5, 5]
prepared_gifts_2 = prepare_gifts(gifts2)
print(prepared_gifts_2)  # [5, 6]

gifts3 = []
prepared_gifts_3 = prepare_gifts(gifts3)
print(prepared_gifts_3)  # []
# There are no gifts, the list remains emp

File: 2024/javascript/01-firstGiftRepeated.js
// Challenge #1: 🎁 First Gift Repeated!

/**
 * @param {number[]} gifts - The array of gifts to prepare
 * @returns {number[]} An array with the prepared gifts
 */
function prepareGifts(gifts) {
  return [...new Set(gifts)].sort((a, b) => a - b);
}

const gifts1 = [3, 1, 2, 3, 4, 2, 5];
const preparedGifts1 = prepareGifts(gifts1);
console.log(preparedGifts1); // [1, 2, 3, 4, 5]

const gifts2 = [6, 5, 5, 5, 5];
const preparedGifts2 = prepareGifts(gifts2);
console.log(preparedGifts2); // [5, 6]

const gifts3 = [];
const preparedGifts3 = prepareGifts(gifts3);
console.log(preparedGifts3); // []
// There are no gifts, the list remains empty
